src/server/pegasus_server_impl.cpp

```diff
@@ -755,11 +784,21 @@ void pegasus_server_impl::on_multi_get(const ::dsn::apps::multi_get_request &req
 
         std::unique_ptr<rocksdb::Iterator> it;
         bool complete = false;
+        uint64_t iteration_time = dsn_now_ns();
+        bool exceed_limit = false;
         if (!request.reverse) {
             it.reset(_db->NewIterator(_data_cf_rd_opts));
             it->Seek(start);
             bool first_exclusive = !start_inclusive;
-            while (count < max_kv_count && size < max_kv_size && it->Valid()) {
+            while (iteration_count < max_iteration_count && size < max_iteration_size &&
+                   it->Valid()) {
+                iteration_time = dsn_now_ns();
```

Comment:
Call `dsn_now_ns` for each iteration may be to costly?

Reply:
I consider your comment and I agree that it is too costly. Most normal iteration won't cost such long time, we needn't check every iteration, so I update it, check at most 10 times during iteration. For example, if max_iteration_count is 100, I only check if time exceed when iteration_count % 10 == 0.